app/src/main/java/com/example/campusguide/search/amenities/AmenitiesSearchResultProvider.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.campusguide.search.amenities
+
+import com.example.campusguide.search.SearchResult
+import com.example.campusguide.search.SearchResultProvider
+
+class AmenitiesSearchResultProvider : SearchResultProvider {
+
+    final val bathroom = "bathroom"
+    final val women = "women's bathroom"
+    final val men = "men's bathroom"
+    final val unisex = "unisex bathroom"
+
+    override suspend fun search(query: String): List<SearchResult> {
+        if (query.isEmpty()) {
+            return emptyList()
+        }
+        val results: MutableList<SearchResult> = mutableListOf()
+        if (bathroom.startsWith(query.toLowerCase())) {
+            results.add(
+                SearchResult(
+                    "Women's Bathroom", "Find a bathroom near me", "amenities_bath_wom"
+                )
+            )
+            results.add(
+                SearchResult(
+                    "Men's Bathroom", "Find a bathroom near me", "amenities_bath_mens"
+                )
+            )
+            results.add(
+                SearchResult(
+                    "Unisex Bathroom", "Find a bathroom near me", "amenities_bath_all"
```

Comment:
There's a lot of repeating yourself code-wise in this function. You should create individual functions for each type of search result, instead of repeating that same block of code every time in each of the different if/else blocks. In addition, each of those strings you're passing when creating a `SearchResult` should be its own string variable, preferably stored in the Constants file.

Reply:
good idea! thanks :)